modules/agent/pkg/controllers/bundledeployment/controller.go

```diff
@@ -213,8 +232,12 @@ func (h *handler) MonitorBundle(bd *fleet.BundleDeployment, status fleet.BundleD
 	readyError := readyError(status)
 	condition.Cond(fleet.BundleDeploymentConditionReady).SetError(&status, "", readyError)
 	if len(status.ModifiedStatus) > 0 {
-		h.bdController.EnqueueAfter(bd.Namespace, bd.Name, 5*time.Minute)
-		if shouldRedeploy(bd) {
+		duration := customResyncDuration(bd)
+		h.bdController.EnqueueAfter(bd.Namespace, bd.Name, duration)
+		ok := shouldRedeploy(bd, &status)
+		if ok {
+			// manage resync counter
+			status.ResyncCounter = manageResyncCounter(bd)
```

Comment:
Can this method be done in `DeployBundle` method? Basically we only set `ResyncCounter` after deploying it to possibly avoid the situation where this is set twice.

Reply:
Does make sense. I have moved it into `DeployBundle`